api/src/OpenTelemetry/Internal/Common/Types.hs

```diff
@@ -175,6 +200,25 @@ data FlushResult
   deriving (Show)
 
 
+-- | Returns @FlushError@ if either argument is @FlushError@, @FlushTimeout@ if either argument is @FlushTimeout@, and @FlushSuccess@ otherwise.
+takeWorseFlushResult :: FlushResult -> FlushResult -> FlushResult
```

Comment:
If you take my advice on changing `FlushError`, you can make this into `(<>)` by implementing the `Semigroup` instance:

```haskell
-- | Returns @FlushError@ if either argument is @FlushError@, @FlushTimeout@ 
-- if either argument is @FlushTimeout@, and @FlushSuccess@ otherwise.
instance Semigroup FlushResult where
  (<>) (FlushError l) (FlushError r) = FlushError (l <> r)
  (<>) (FlushError l) _ = FlushError l
  (<>) _ (FlushError r) = FlushError r
  (<>) FlushTimeout _ = FlushTimeout
  (<>) _ FlushTimeout = FlushTimeout
  (<>) FlushSuccess FlushSuccess = FlushSuccess

instance Monoid FlushResult where
  mempty = FlushSuccess
```